src/Matrix1L.java

```diff
@@ -0,0 +1,99 @@
+
+/**
+ * Matrix represented as a 2D array of doubles with
+ * implementations of primary methods.
+ *
+ * @convention
+ * [All rows in this.rep have the same length]
+ * and for all i, j: [rep[i][j] is a valid double]
+ *
+ * @correspondence
+ * this = [matrix with rows = rep.length, columns = rep[0].length,
+ *         where entry (i, j) = rep[i][j]]
+ */
+public class Matrix1L extends MatrixSecondary {
+
+    /**
+     * Representation of {@code this}.
+     */
+    private double[][] rep;
+
+    /**
+     * Creator of initial representation.
+     */
+    private void createNewRep(int rows, int cols) {
+        this.rep = new double[rows][cols];
+    }
+
+    /**
+     * No-argument constructor.
+     */
+    public Matrix1L() {
+        this.createNewRep(3, 3);
+    }
+
+    /**
+     * Constructor from dimensions.
+     *
+     * @param rows
+     *            number of rows
+     * @param cols
+     *            number of columns
+     */
+    public Matrix1L(int rows, int cols) {
+        assert rows >= 0 : "Violation of: rows >= 0";
+        assert cols >= 0 : "Violation of: cols >= 0";
+
+        this.createNewRep(rows, cols);
+    }
+
+    @Override
+    public final Matrix newInstance() {
+        return new Matrix1L(this.rows(), this.columns());
```

Comment:
This probably works, but I would mirror your solution to the solutions you see in previous projects. Same goes for clear and transferFrom.